tealish/nodes.py

```diff
@@ -63,17 +77,24 @@ def add_child(self, node):
         self.child_nodes.append(node)
 
     @classmethod
-    def consume(cls, compiler, parent):
+    def consume(cls, compiler: "TealishCompiler", parent: Optional["Node"]) -> "Node":
         line = compiler.consume_line()
         return cls(line, parent=parent, compiler=compiler)
 
-    def write(self, lines):
+    def write(self, lines: List[str]) -> None:
+        if self.compiler is None:
+            raise Exception("Can't write to a compiler when its none??")
+
         self.compiler.write(lines, self.line_no)
 
-    def get_current_scope(self):
+    def get_current_scope(self) -> Dict[str, Any]:
+        if self.current_scope is None:
+            raise Exception("No scope???")
```

Comment:
Shouldn't stay but an Optional (maybe none) on the scope would complain otherwise